scripts/check_compliance.py

```diff
@@ -588,7 +589,16 @@ def run(self):
 
                 original_fp = str(file['path']).replace('scancode-files/', '')
                 licenses = file['licenses']
-                if (file['is_script'] or file['is_source']) and (file['programming_language'] not in whitelist_languages) and (file['extension'] not in whitelist_extensions):
+                is_script = file.get("is_script")
+                is_source = file.get("is_source")
+                is_media = file.get("is_media")
+                pl = file.get("programming_language", None)
+                extension = file.get("extension")
+                file_type = file.get("file_type")
+                kconfig = "Kconfig" in original_fp and file_type in ['ASCII text']
```

Comment:
Had to read back to understand that "fp" stands for "file path". Could call it `original_file_path` instead maybe, or just `original_path`.